core/src/main/java/io/github/wysohn/triggerreactor/trblock/main/TRBlockManager.java

```diff
@@ -0,0 +1,68 @@
+package io.github.wysohn.triggerreactor.trblock.main;
+
+import io.github.wysohn.triggerreactor.core.manager.Manager;
+import io.javalin.Javalin;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.logging.Logger;
+
+@Singleton
+public class TRBlockManager {
```

Comment:
Looks really good

But I doubt if having TRBlockManager and TRBlock would make sense since they seem to be doing closely related jobs

If the concern is executing `public boolean start()` only once, we could put that into `initialize()` directly instead.

If you prefer having multiple layers of abstraction (which is good), consider creating TRBlock as an interface and creating the `TRBlockImpl` class which implements the TRBlock interface. In this way, we can avoid depending directly on Javalin. Or it doesn't have to be `TRBlock` but some more generic name like `TRBlockWebServiceProvider`

Reply:
i got it